### ISSUE TITLE
Unexpected token error caused by different versions of flow

### ISSUE THREAD

**Issue by CodyReichert** [amilajack/eslint-plugin-flowtype-errors#43]
If I have the following type defined inside of a react component. This is using covariant properties in an exact object type:

```
state : {|
    +loading : boolean,
|}
```

I get an error from this plugin:

```
Unexpected token +: [flowtype-errors/show-errors]
```

However, this is valid syntax ([see here](https://flowtype.org/try/#0C4TwDgpgBAChBOBnA9gOygXigbwD4CgooBqAM1QEMBbCALikWHgEtUBzAGkJIBtKb6jFuy5FiAV0QJ+dBk1Zt8uAL4BufPh4RgUeONSoAjO3pwkaTDm7lqsgEQBpCvAg87oqH1v07AIQC3WgDW7tyS0t5QdnoGxmyhahoA9ElQAJLAAOSIUMhBUMDIuhAUACZQTlAAxsgAbs7MFKg6YPDIkPCg+AAUMUbsAHQ2AnLCbACU6vgpUACi8G3wAIRQAMJNmToA7izA0IVQFNV1DU0tbR1dKX1xQzKWdqjtEHaqICBAA)).

And flow does not throw any errors.

I'm assuming this is because this plugin has a dependency on `flow-bin@^0.33.0` whereas my project is using `flow-bin@^0.35.0`.

Can this plugin be configured to use _my_ flow-bin? Or can the restraints on the dependency somehow be loosened?

**Comment by CodyReichert** [comment]
Semi-related now regarding:

> Can this plugin be configured to use my flow-bin?

I see in in a few places that the plugin checks for `process.env.FLOW_BIN`, and if that's not found, it `require`s `flow-bin`. (eg, https://github.com/amilajack/eslint-plugin-flowtype-errors/blob/master/src/collect.js#L16)

What would be the best way to _set_ `process.env.FLOW_BIN` so that the plugin would use my version of flow? Is that possible?

**Comment by CodyReichert** [comment]
Last bit of info - just realized I'm on `eslint-plugin-flowtype-errors@1.5.0`, but `2.0.0` makes some changes to the `flow-bin` dependency that might fix this.

Can a new version be released to npm?

**Comment by amilajack** [comment]
Just to confirm, try this:

```bash
npm cache clean
rm -rf node_modules
npm i -D eslint-plugin-flowtype-errors
npm i -D flow-bin
```

**Comment by CodyReichert** [comment]
@amilajack - thanks. Same error, unfortunately:

```
Unexpected token +: [flowtype-errors/show-errors]
```

Note that the `+` syntax was introduced in flow@0.34.0, and this plugin has a dependency on version 0.33.0.

---

The `flow-bin` in my top-level `node_modules/.bin` directory is the latest 0.36.0 version:

```
λ ./node_modules/.bin/flow version
Flow, a static type checker for JavaScript, version 0.36.0
```

And the `flow-bin` in the `node_modules` for this plugin is `0.33.0`:

```
λ ./node_modules/eslint-plugin-flowtype-errors/node_modules/.bin/flow version
Flow, a static type checker for JavaScript, version 0.33.0
```

I'm assuming this plugin is using the 0.33.0 instead of 0.36.0.


Here's a screenshot, if it helps:

![eslint-plugin-flowtype-errors-error](https://cloud.githubusercontent.com/assets/7034627/20902226/ceea0354-bafb-11e6-8b56-d6f1b2bf755f.png)

**Comment by amilajack** [comment]
Ahh I see. Thanks. I'll update

**Comment by amilajack** [comment]
Sorry for the delay. I've just pushed some fixes for this in a preminor:

```bash
npm i -D eslint-plugin-flowtype-errors@2.0.1-0
```

**Comment by CodyReichert** [comment]
Just tested it and the error no longer appears -- thank you!
